tests/PSRule.Rules.Azure.Tests/Azure.Template.Tests.ps1

```diff
@@ -655,5 +655,39 @@ Describe 'Azure.Template' -Tag 'Template' {
             $targetNames = $ruleResult | ForEach-Object { $_.TargetName.Split([char[]]@('\', '/'))[-1] };
             $targetNames | Should -BeIn 'Resources.Parameters.json', 'Resources.VPN.Parameters3.json';
         }
+
+        It 'Azure.Template.UseComments' {
+            $dataPath = Join-Path -Path $here -ChildPath 'Resources.Template*.json';
+            $result = Invoke-PSRule @invokeParams -InputPath $dataPath -Format None -Name 'Azure.Template.UseComments';
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.Template.UseComments' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 3;
+            $ruleResult.TargetName | Should -BeLike "*Resources.Template[2-4].json";
+
+            $ruleResult[0].Reason | Should -HaveCount 2;
+            $ruleResult[0].Reason[0] | Should -BeExactly "The field 'comments' does not exist.";
+            $ruleResult[0].Reason[1] | Should -BeExactly "The field 'comments' does not exist.";
+
+            $ruleResult[1].Reason | Should -HaveCount 4;
+            $ruleResult[1].Reason[0] | Should -BeExactly "The field 'comments' does not exist.";
+            $ruleResult[1].Reason[1] | Should -BeExactly "The field 'comments' does not exist.";
+            $ruleResult[1].Reason[2] | Should -BeExactly "The field 'comments' does not exist.";
+            $ruleResult[1].Reason[3] | Should -BeExactly "The field 'comments' does not exist.";
```

Comment:
Was seeing if we could put resource name in the reasons, but this doesn't work out when we use ARM template expressions like `[concat(parameters('vnetName'), '/', parameters('clusterSubnet'), '/Microsoft.Authorization/', guid(parameters('clusterSubnet')))]`, since that is what gets shown in the reason if I do something like `.Reason($LocalizedData.UseComments, $resource.name)` in the rule.

Reply:
> Strange how the tests are failing for macOS.

This likely the order the files are being processed, which is different from Windows. Filter `$ruleResult` by target name before you compare.

Reply:
At the moment there isn't an easy way to solve this by looking at the template directly. You could potentially do it based on line number using source info since you are calling via $PSRule.GetContent. 

https://github.com/microsoft/PSRule/blob/1a74b5afd1c07053695b470c8559279430374378/src/PSRule/Pipeline/PipelineReciever.cs#L330

Another approach would be to look at the resources. Currently the comments field is removed but it could be emitted. This would give you a resolved resource name. The `_PSRule` meta property automatically gets read by doesn't have a generic dictionary that we can add anything to. You could emit an issue though similar to https://github.com/Azure/PSRule.Rules.Azure/blob/ae50c78797168fa4bd23539e2bafa4a3ed684a4b/src/PSRule.Rules.Azure/rules/Azure.Template.Rule.ps1#L235.